src/mail-app/mail/model/MailUtils.ts

```diff
@@ -64,13 +65,19 @@ export async function getMoveTargetFolderSystems(foldersModel: MailModel, mails:
 		return []
 	}
 
-	return folders.getIndentedList().filter((f: IndentedFolder) => {
-		if (f.folder.isMailSet && isNotEmpty(firstMail.sets)) {
-			return !isSameId(f.folder._id, folder._id)
-		} else {
-			return f.folder.mails !== getListId(firstMail)
-		}
-	})
+	const isFolderView = foldersModel.isFolderView && !m.route.get().startsWith("/search")
```

Comment:
I am also very confused about what's going on here

Reply:
This also has been changed to check if the selected mails are in different folders, which is a cleaner approach to determining whether or not to show all the move targets.